src/configurator.rb

```diff
@@ -30,6 +30,43 @@
 
 module Metalware
   class Configurator
+    class << self
+      def for_domain(file_path:)
```

Comment:
These methods shouldn't really have `file_path` passed into them. Currently this is fine because only the `file_path` is required, however in the `Configurator` will need the `config` so it can create the file `loader`.

So to prevent future conflicts, the `config` should be passed into the methods and then each method creates it's own `FilePath` object based on the `config`. This will temporarily add some overhead. However this will allow me to refactor `Configurator` (and move `FilePath` inside it) without having to make any changes to the `GUI`.